cairo/tests/fixtures/runner.py

```diff
@@ -58,39 +59,56 @@ def cairo_run(request, cairo_program, cairo_file, main_path):
     """
 
     def _factory(entrypoint, *args, **kwargs):
-        implicit_args = list(
-            cairo_program.identifiers.get_by_full_name(
-                ScopedName(path=("__main__", entrypoint, "ImplicitArgs"))
-            ).members.keys()
-        )
+        implicit_args = cairo_program.identifiers.get_by_full_name(
+            ScopedName(path=("__main__", entrypoint, "ImplicitArgs"))
+        ).members.items()
+
+        # Split implicit args into builtins and other implicit args
+        _builtins = {
+            k: {
+                "python_type": to_python_type(
+                    resolve_main_path(main_path)(v.cairo_type)
+                ),
+                "cairo_type": v.cairo_type,
+            }
+            for k, v in implicit_args
+            if any(builtin in k.replace("_ptr", "") for builtin in ALL_BUILTINS)
+        }
+
+        _implicit_args = {
+            k: {
+                "python_type": to_python_type(
+                    resolve_main_path(main_path)(v.cairo_type)
+                ),
+                "cairo_type": v.cairo_type,
+            }
+            for k, v in implicit_args
+            if not any(builtin in k.replace("_ptr", "") for builtin in ALL_BUILTINS)
+        }
```

Comment:
```suggestion
        implicit_args = cairo_program.identifiers.get_by_full_name(
            ScopedName(path=("__main__", entrypoint, "ImplicitArgs"))
        ).members

        # Split implicit args into builtins and other implicit args
        _builtins = {
            k.replace("_ptr", "")
            for k in implicit_args.values()
            if any(builtin in k.replace("_ptr", "") for builtin in ALL_BUILTINS)
        }

        _implicit_args = {
            k: {
                "python_type": to_python_type(
                    resolve_main_path(main_path)(v.cairo_type)
                ),
                "cairo_type": v.cairo_type,
            }
            for k, v in implicit_args.items()
            if not any(builtin in k.replace("_ptr", "") for builtin in ALL_BUILTINS)
        }
```

Reply:
not replacing the "ptr" in _builtins here with an empty string, as it further complexifies the process of replacing the _ptr suffix by `builtin` when handling builtins down the line